### ISSUE TITLE
Authentication not working

### ISSUE THREAD

**Issue by insanity67** [CorentinTh/enclosed#181]
### Describe the bug

I installed enclosed, everything works perfectly without authentication.

I deleted the conatiner and volume, and recreated with following docker line:
`docker run --name enclosed --restart unless-stopped -p 8787:8787 -e PUBLIC_IS_AUTHENTICATION_REQUIRED="true" -e AUTHENTICATION_JWT_SECRET="cff8d041963fcd1330xxxxxxxxxxxxe36b4677663651afa52eca9855" -e AUTHENTICATION_USERS=myuser@mydomain.com:$2a$10$yMW8mF.L2Na68/xnAoqrXej219jrYcU89dWfe/tfKbtyOZLZ.B1sa -e SERVER_CORS_ORIGINS="https://enc.mydomain.com" -v enclosed_data:/app/.data corentinth/enclosed`

The AUTHENTICATION_USERS string was generated with: https://docs.enclosed.cc/self-hosting/users-authentication-key-generator

But I cannot login, tells me everytime "wrong username/password".

This logs did I get from docker:
```
{"level":30,"time":1726736855966,"pid":1,"hostname":"5d0f7c09b27d","namespace":"app","status":200,"method":"GET","path":"/login","routePath":"/*","durationMs":7,"msg":"Request completed"}
{"level":30,"time":1726736855999,"pid":1,"hostname":"5d0f7c09b27d","namespace":"app","status":200,"method":"GET","path":"/api/config","routePath":"/api/config","durationMs":1,"msg":"Request completed"}
{"level":30,"time":1726736856051,"pid":1,"hostname":"5d0f7c09b27d","namespace":"app","status":200,"method":"GET","path":"/site.webmanifest","routePath":"/*","durationMs":1,"msg":"Request completed"}
{"level":50,"time":1726736861769,"pid":1,"hostname":"5d0f7c09b27d","namespace":"middlewares:error","error":{"code":"auth.unauthorized","cause":{},"statusCode":401,"isCustomError":true},"msg":"Unauthorized"}
{"level":30,"time":1726736861770,"pid":1,"hostname":"5d0f7c09b27d","namespace":"app","status":401,"method":"POST","path":"/api/auth/login","routePath":"/api/auth/login","durationMs":5,"msg":"Request completed"}
```

### What happened?

I expected to log in with my created username/password.

### System information

Docker version 27.2.1, build 9e34c9b on Debian 12

### Where did you encounter the bug?

Other (installations, docker, etc.)

**Comment by insanity67** [comment]
I even tried enclosing AUTHENTICATION_USERS value in "" but did not change the error message.

**Comment by Bartrpc** [comment]
+1 
Same problem here. I suspect that this could be related to _middleware_ , which reverse proxy are you using ?
I'm using Cloudflare tunnels and I supose it's  not supported.

**Comment by insanity67** [comment]
> which reverse proxy are you using ?

Im using nginx proxy manager with pretty much default settings:

![image](https://github.com/user-attachments/assets/62dc10b1-cff9-435d-9ba7-4e1838ab9574)

![image](https://github.com/user-attachments/assets/fc3c9e31-ca9b-4bfd-9812-ffee41813cc8)

**Comment by Bartrpc** [comment]
@CorentinTh am I right, that some reverse proxy are not supported yet ?

**Comment by insanity67** [comment]
@Bartrpc hmm... but how to self host this with ssl without some kind of reverse proxy? 🤔

**Comment by Bartrpc** [comment]
it must work behind reverse proxy, there is a pop-up with info that if you're using Enclosed with HTTP you will not be able to create notes.
I dunno what to do. For now I'm stopping container and waiting for developer response ;)

**Comment by CorentinTh** [comment]
Hi @Bartrpc 

When passing environment variables in the Docker CLI using the `-e` flag, you need to escape the `$` sign (i.e., replace `$` with `\$`). This is because the `$` sign is interpreted by bash.

For example, with your command:

```bash
docker run --name enclosed --restart unless-stopped -p 8787:8787 -e PUBLIC_IS_AUTHENTICATION_REQUIRED="true" -e AUTHENTICATION_JWT_SECRET="cff8d041963fcd1330xxxxxxxxxxxxe36b4677663651afa52eca9855" -e AUTHENTICATION_USERS="myuser@mydomain.com:\$2a\$10\$yMW8mF.L2Na68/xnAoqrXej219jrYcU89dWfe/tfKbtyOZLZ.B1sa" -e SERVER_CORS_ORIGINS="https://enc.mydomain.com/" -v enclosed_data:/app/.data corentinth/enclosed
```

Alternatively, you can use the `--env-file` option to load environment variables from a file without needing to escape the content

I'll also consider adding an option to pass this variable as base64 in the future, which would eliminate the need to escape special characters.

**Comment by Bartrpc** [comment]
Still nothing, I'm using docker compose btw and deploying enclosed via dockge :) 

![screen-enclosed](https://github.com/user-attachments/assets/5b2c1207-45f6-48f3-8a9d-5fa5fb519e8d)

**Comment by CorentinTh** [comment]
No escaping needed in .env file

**Comment by insanity67** [comment]
@CorentinTh Wow, thanks for your quick reply!
I escaped the "$", not everything works as expected 😍

**Comment by CorentinTh** [comment]
I'll add a warning on the variable generator page

**Comment by insanity67** [comment]
@CorentinTh good idea, thank you! Really love that project, was looking for such a solution since ever.

**Comment by Bartrpc** [comment]
Ok, now works, thank You. 
It turns out that in **env_file** I had to do it with "" and "\" and it's working :) 
.env -> `AUTHENTICATION_USERS=bartoszrpc@proton.me:$2a$10$8M4JjO8YP6pkDCqtFolzUOSb25ptxA3qqVCoa7ne50SfmrzUWQN5C`

**Comment by Bartrpc** [comment]
You've to escape $ in .env file :) It's mandatory :)

**Comment by Bartrpc** [comment]
Thank You @CorentinTh - fantastic idea with that project :)
